packages/admin/cms-admin/src/generator/future/generateFormField.ts

```diff
@@ -106,12 +113,91 @@ export function generateFormField(
+
+        //find labelField: 1. as configured
+        let labelField = config.labelField;
+
+        //find labelField: 2. common names (name or title)
+        if (!labelField) {
+            labelField = objectType.fields.find((field) => {
+                let type = field.type;
+                if (type.kind == "NON_NULL") type = type.ofType;
+                if ((field.name == "name" || field.name == "title") && type.kind == "SCALAR" && type.name == "String") {
+                    return true;
+                }
+            })?.name;
+        }
+
+        //find labelField: 3. first string field
+        if (!labelField) {
+            labelField = objectType.fields.find((field) => {
+                let type = field.type;
+                if (type.kind == "NON_NULL") type = type.ofType;
+                if (field.type.kind == "SCALAR" && field.type.name == "String") {
+                    return true;
+                }
+            })?.name;
+        }
+
+        const rootQuery = config.rootQuery; //TODO we should infer a default value from the gql schema
+        const queryType = objectType.name;
+        const queryVariableName = `${rootQuery}Query`;
+        const queryName = rootQuery[0].toUpperCase() + rootQuery.substring(1);
+        const fragmentVariableName = `${rootQuery}SelectFragment`;
+        const fragmentName = `${objectType.name}Select`;
+
+        formFragmentField = `${name} { id ${labelField} }`;
+
+        gqlDocuments[fragmentVariableName] = `
+            fragment ${fragmentName} on ${queryType} {
+                id
+                ${labelField}
+            }
+        `;
+        gqlDocuments[queryVariableName] = `query ${queryName} {
+            ${rootQuery} {
+                nodes {
+                    ...${fragmentName}
+                }
+            }
+        }
+        \${${fragmentVariableName}}
+        `;
+
+        imports.push({
+            name: "useAsyncOptionsProps",
+            importPath: "@comet/admin",
+        });
+        hooksCode += `const ${name}SelectAsyncProps = useAsyncOptionsProps(async () => {
+            const result = await client.query<GQL${queryName}Query, GQL${queryName}QueryVariables>({ query: ${queryVariableName} });
+            return result.data.${rootQuery}.nodes;
+        });`;
+
+        formValueToGqlInputCode = `${name}: formValues.${name}?.id,`;
+
+        code = `<Field
+                fullWidth
+                name="${name}"
+                label={<FormattedMessage id="${instanceGqlType}.${name}" defaultMessage="${label}" />}
+                component={FinalFormSelect}
+                {...${name}SelectAsyncProps}
+                getOptionLabel={(option: GQL${fragmentName}Fragment) => option.${labelField}}
+            />`;
     } else {
-        throw new Error(`Unsupported type: ${config.type}`);
+        throw new Error(`Unsupported type`);
```

Comment:
why did you remove the type-info?

Reply:
because ts is so clever that is knows that else can't be reached and config is of type never.

I still wanted to keep the else for the case that future types get added